sdk/core/azure-core/azure/core/utils/_pipeline_transport_rest_shared.py

```diff
@@ -344,16 +345,31 @@ def _format_data_helper(data: Union[str, IO]) -> Union[Tuple[None, str], Tuple[O
     :rtype: tuple[str, IO, str] or tuple[None, str]
     :return: A tuple of (data name, data IO, "application/octet-stream") or (None, data str)
     """
-    if hasattr(data, "read"):
+    content_type = "application/octet-stream" # default content type
```

Comment:
How about leaving the method as it is and creating a new one _format_tuple_data_helper to handle the tuple case?

Reply:
I think it's cleaner if we leave it as one because we just return one tuple at the end of `filename`, `file_bytes`, and `content_type`. Open to changing though if you feel strongly